CDCgov/prime-reportstream
# Changes PID-13 repeat order so phone number comes first

This PR inverts the PID-13 repeat order so that phone number comes first.

In HL7, the PID-13 field is not just for patient phone number. It is for all kinds of telecommunication devices including pagers, faxes, and emails. The convention is that the patient's home phone number comes first in the list, and we have not being doing that. The reason we haven't been doing this is because when we walk through the fields in our internal format for a record, `patient_email` comes before `patient_phone`, which means that it gets written first. Because this breaks convention, and many systems can't process anything in PID-13 beyond the first repeat, we built in a hack to move the `patient_email` to PID-14. That has held us over until now when we are encountering states that don't want to accept a value in PID-14. In truth, this is something we should have fixed a while ago anyway.

The new logic does an examination of the `/PATIENT_RESULT/PATIENT/PID-13(0)` field and looks to see if it contains an email address (where `(0)` means first repeat in HL7 paths). If it does, and if we are writing a patient phone number, we now extract out the email address and overwrite the `(0)` repeat in `/PATIENT_RESULT/PATIENT/PID-13(0)` with the phone number data, and then write out the email to the second repeat.

This does make an assumption that for a patient we only have a maximum of TWO values that could go into PID-13: home phone and email. We don't currently accept any other device numbers or alternate numbers for patients. There is a chance we will need to revisit this logic in the future and write it to handle multiple phone numbers for a patient.

Test Steps:
1. Create a sample CSV file that contains three records: one with just an email, one with just a phone number, and one with an email and a phone number.
2. Submit this file to your docker container and let it go through the ingestion process
3. Inspect the file that was created by your Docker container and verify that for the cases of just an email and just a phone number, there is no repeat in PID-13 and the email or phone number are passed through as expected. For the patient record with both an email and a phone number, observe that the patient phone number is first in the PID-13 field, which is followed by a tilde `~` and then the patient email 

## Changes
- Updated the HL7 phone logic to push the email to repeat 2
- Added a unit test to make sure the terser could and properly did handle get and setting to multiple repeats

## Checklist

### Testing

### Process

## Fixes
- #2214
